ZenPacks/zenoss/OpenStackInfrastructure/events.py

```diff
@@ -19,11 +19,38 @@
                                                            get_port_fixedips,
                                                            is_uuid,
                                                            )
+from Products.ZenModel.ZVersion import VERSION as ZENOSS_VERSION
+from distutils.version import LooseVersion
+
 import ast
 
 import logging
 LOG = logging.getLogger('zen.OpenStack.events')
 
+# Starting from zenoss 6.4.0 ADM code were reworked without full compatibility 
+# with lefacy directives provided by PythonCollector zenpack, add some tweaks here 
+# to make directives work with old and new ADM
+NEW_ADM = '6.4.0'
```

Comment:
This is not strictly compatible.  All versions of CZ will assume new ADM code.  Although this may not be an issue in practice since, in theory, all Cloud deployments use a recent release.

A detection mode that does not rely on versioning is as follows:
```
try:
    import Products.DataCollector.ApplyDataMap
except ImportError:
    NEW_ADM = False
else:
    NEW_ADM = True

...

if NEW_ADM:
    # new adm stuff
else:
    # old adm stuff
```

This will work because the `Products.DataCollector.ApplyDataMap` package only exists in RM/CZ systems that have the new ADM code.